modules/java/generator/src/cpp/common.h

```diff
@@ -16,6 +16,20 @@ extern "C" {
 #endif
 #include <jni.h>
 
+// make -fvisibility=hidden work with java 1.7
+#if defined(__linux__) && !defined (JNI_VERSION_1_8)
+  // adapted from jdk1.8/jni.h
+  #if (defined(__GNUC__) && ((__GNUC__ > 4) || (__GNUC__ == 4) && (__GNUC_MINOR__ > 2))) || __has_attribute(visibility)
+    #undef  JNIEXPORT
+    #define JNIEXPORT     __attribute__((visibility("default")))
+    #undef  JNIIMPORT
+    #define JNIIMPORT     __attribute__((visibility("default")))
```

Comment:
> JNIIMPORT

Looks like OpenCV doesn't use `JNIIMPORT`.

Reply:
It's from Oracle 7u25, which we have to use for a customer of ours :-(
jdk1.7.0_25/include/linux/jni_md.h

Reply:
I added JNIIMPORT to have it consistent to JNIEXPORT.

Reply:
I mean that `JNIIMPORT` changes are not needed here (due `JNIIMPORT` is not used).

Reply:
My idea is that even if JNIIMPORT is not used now, the first one to use it would run into the same trouble I had with JNIEXPORT. The default definition of JNIIMPORT is as wrong as that of JNIEXPORT with JDK1.7. So I would like to have them both fixed at once.